pkg/prometheus/server/operator.go

```diff
@@ -951,6 +962,22 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 	return nil
 }
 
+// As the ShardRetentionPolicy feature evolves, should delete will evolve accordingly.
+// For now, shouldDelete just returns the appropriate boolean based on the retention type.
+func (c *Operator) shouldDelete(p *monitoringv1.Prometheus) (bool, error) {
+	if !c.retentionPoliciesEnabled {
+		// Feature-gate is disabled, default behavior is always to delete.
+		return true, nil
+	}
+
+	if p.Spec.ShardRetentionPolicy.WhenScaled != nil &&
```

Comment:
I think this work as well.

```suggestion
	if ptr.Dref(p.Spec.ShardRetentionPolicy.WhenScaled, monitoringv1.DeleteWhenScaledRetentionType) == monitoringv1.DeleteWhenScaledRetentionType {
	return true, nil
}
```